tmk_core/protocol/report.h

```diff
@@ -196,6 +197,11 @@ typedef struct {
     uint16_t usage;
 } __attribute__((packed)) report_extra_t;
 
+typedef struct {
+    uint8_t report_id;
+    uint8_t state;
+} __attribute__((packed)) report_radio_t;
```

Comment:
There is no need for most of this duplicate report sending code - you can use `report_extra_t`/`send_extra()` etc. and let the host differentiate based on the report ID.

Reply:
The size doesn't need to be the same if the report ID is different - this is how the shared endpoint works.

Reply:
But in that case `send_extra()` would need to examine the report ID to determine the appropriate USB packet size; currently it just uses `sizeof(report_extra_t)`.

Reply:
That would work, just need to change the number of padding bits in the report from 7 to 15, and then doing something like `.usage = (uint16_t)state` when initializing the report.